app/models/measurement_details.rb

```diff
@@ -21,12 +21,17 @@ def initialize(attributes = {})
   def load
     validate!
 
-    load_total_from_gr
-    load_local_from_gr
-    load_user_from_gr
-    load_sub_mins_from_gr
-    load_team_from_gr
-    load_split_measurements
+    # make sure instance vars are set
+    @current_power = Power.current
+    gr_singleton
+    ministry
+
+    threads = [:load_local_from_gr, :load_total_from_gr, :load_user_from_gr, :load_sub_mins_from_gr,
+               :load_team_from_gr, :load_split_measurements]
+    threads = threads.map do |method|
+      Thread.new { send(method) }
+    end
+    threads.each { |t| t.join(1) }
```

Comment:
As I think about it, it will give 1 second per thread, for a total of 6 seconds or more which feels like a lot of time for it to finish. But still, I it would be clearer either to think about how you would handle a timeout if it occurs or just to not include the timeout.